src/fastfield/multivalued/reader.rs

```diff
@@ -27,24 +27,30 @@ impl<Item: FastValue> MultiValuedFastFieldReader<Item> {
         }
     }
 
-    /// Returns `(start, stop)`, such that the values associated
-    /// to the given document are `start..stop`.
+    /// Returns `[start, end)`, such that the values associated
+    /// to the given document are `start..end`.
     #[inline]
     fn range(&self, doc: DocId) -> Range<u64> {
         let start = self.idx_reader.get(doc);
-        let stop = self.idx_reader.get(doc + 1);
-        start..stop
+        let end = self.idx_reader.get(doc + 1);
+        start..end
     }
 
     /// Returns the array of values associated to the given `doc`.
     #[inline]
-    pub fn get_vals(&self, doc: DocId, vals: &mut Vec<Item>) {
-        let range = self.range(doc);
+    fn get_vals_for_range(&self, range: Range<u64>, vals: &mut Vec<Item>) {
         let len = (range.end - range.start) as usize;
```

Comment:
I think there is a range.len()

Reply:
weirdly doesn't seem to be implemented for Range<u64>

Reply:
It's given via the [ExactSizeIterator](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html) trait